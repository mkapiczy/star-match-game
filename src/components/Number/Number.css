.number {
    margin-right: 5px;
    background-color: #aeaeae;
    border: none;
    color: white;
    text-align: center;
    text-decoration: none;
    border-radius: 50%;
    width: 5vmin;
    height: 5vmin;
}

.number.not-available {
    background-color: #3F51B5;
}

.number:not(.not-available):hover {
    background-color: #d7d7d7;
}