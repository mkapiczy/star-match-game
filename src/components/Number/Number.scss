.number {
  margin-right: 0.25em;
  margin-left: 0.25em;
  background-color: #aeaeae;
  border: none;
  color: white;
  text-align: center;
  text-decoration: none;
  border-radius: 50%;
  width: 5vmin;
  height: 5vmin;
  font-size: 2.5vmin;

  &.not-available {
    background-color: #3F51B5;
  }

  &:not(.not-available):hover {
    background-color: #d8d8d8;
  }
}