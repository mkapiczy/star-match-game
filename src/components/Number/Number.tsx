import React from 'react';
import './Number.scss';
import classNames from "classnames";

interface Props {
    numberValue: number,
    onNumberClick: (numberValue: number) => void;
    isAvailable: boolean
}

const Number: React.FC<Props> = (props: Props) => {
    return (
        <button key={props.numberValue} className={classNames('number', {'not-available': !props.isAvailable})}
                disabled={!props.isAvailable} onClick={() => props.onNumberClick(props.numberValue)}>
            {props.numberValue}
        </button>
    )
}

export default Number;