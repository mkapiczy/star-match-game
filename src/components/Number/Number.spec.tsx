import React from 'react';
import Number from "./Number";
import {fireEvent, render} from '@testing-library/react';
import renderer from 'react-test-renderer';

describe("Number component tests", () => {
    it("should render Number component", () => {
        const onNumberClick = jest.fn()
        const {getByText} = render(
            <Number numberValue={2} isAvailable={true} onNumberClick={onNumberClick}/>,
        );
        const numberButon = getByText("2")
        expect(numberButon).toBeInTheDocument()
    })
    it("should call onNumberClick method when number button available", ()=> {
        const onNumberClick = jest.fn()
        const {getByText} = render(
            <Number numberValue={2} isAvailable={true} onNumberClick={onNumberClick}/>,
        );
        const numberButon = getByText("2")
        fireEvent.click(numberButon)
        expect(onNumberClick).toBeCalledTimes(1)
    })
    it("should not call onNumberClick method when number button is not available", ()=> {
        const onNumberClick = jest.fn()
        const {getByText} = render(
            <Number numberValue={2} isAvailable={false} onNumberClick={onNumberClick}/>,
        );
        const numberButon = getByText("2")
        fireEvent.click(numberButon)
        expect(onNumberClick).not.toBeCalled()
    })
    it("should match snapshot for button available true", () => {
        const component = renderer.create(
            <Number numberValue={2} isAvailable={true} onNumberClick={() => ''}/>,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
    it("should match snapshot for button available false", () => {
        const component = renderer.create(
            <Number numberValue={2} isAvailable={false} onNumberClick={() => ''}/>,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})