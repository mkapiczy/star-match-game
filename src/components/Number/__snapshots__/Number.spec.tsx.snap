// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Number component tests should match snapshot for button available false 1`] = `
<button
  className="number not-available"
  disabled={true}
  onClick={[Function]}
>
  2
</button>
`;

exports[`Number component tests should match snapshot for button available true 1`] = `
<button
  className="number"
  disabled={false}
  onClick={[Function]}
>
  2
</button>
`;

exports[`Number snapshot tests should match snapshot for button available false 1`] = `
<button
  className="number not-available"
  disabled={true}
  onClick={[Function]}
>
  2
</button>
`;

exports[`Number snapshot tests should match snapshot for button available true 1`] = `
<button
  className="number"
  disabled={false}
  onClick={[Function]}
>
  2
</button>
`;
